library/src/main/scala/scala/scalajs/runtime/package.scala

```diff
@@ -6,6 +6,68 @@ import scala.collection.GenTraversableOnce
 
 package object runtime {
 
+  /** Returns true if we are linking for production, false otherwise.
+   *
+   *  `productionMode` is always equal to `!developmentMode`.
+   *
+   *  This ends up being constant-folded to a constant at link-time. So
+   *  constant-folding, inlining, and other local optimizations can be
+   *  leveraged with this "constant" to write code that should only be
+   *  executed in production mode or development mode.
+   *
+   *  A typical usage of this method is:
+   *  {{{
+   *  val warningsLogger =
+   *    if (productionMode) new NullLogger
+   *    else new ConsoleLogger
+   *  }}}
+   *
+   *  At link-time, `productionMode` will either be a constant true, in which
+   *  case the above snippet folds into
+   *  {{{
+   *  val warningsLogger = new NullLogger
+   *  }}}
+   *  or a constant false, in which case it folds into
+   *  {{{
+   *  val warningsLogger = new ConsoleLogger.
+   *  }}}
+   *
+   *  @see [[developmentMode]]
+   */
+  @inline
+  def productionMode: Boolean =
+    linkingInfo.semantics.productionMode
+
+  /** Returns true if we are linking for development, false otherwise.
+   *
+   *  `developmentMode` is always equal to `!productionMode`.
+   *
+   *  This ends up being constant-folded to a constant at link-time. So
+   *  constant-folding, inlining, and other local optimizations can be
+   *  leveraged with this "constant" to write code that should only be
+   *  executed in production mode or development mode.
+   *
+   *  A typical usage of this method is:
+   *  {{{
+   *  if (developmentMode) {
+   *    performExpensiveSanityChecks()
+   *  }
+   *  }}}
+   *
+   *  At link-time, `developmentMode` will either be a constant true, in which
+   *  case the above snippet folds into
+   *  {{{
+   *  performExpensiveSanityChecks()
+   *  }}}
+   *  or a constant false, in which case it is dead-code-eliminated away,
+   *  yielding maximum performance in production.
+   *
+   *  @see [[productionMode]]
+   */
+  @inline
+  def developmentMode: Boolean =
+    !productionMode
+
```

Comment:
I think we should create a place, since otherwise it is very confusing for users what they may rely upon and what not. For instance, many methods in this package object should definitely not be called in user code. So maybe we need to create `scala.scalajs.LinkingInfo` / `linking_info`?